### DISCUSSION TITLE
Best practice when performing conditions on native fields?

### DISCUSSION THREAD

**Discussion by BenParizek** [craftcms/cms#11359] [Q&A] [answered]
I'm trying to use Conditions on Native tabs and fields. Conditions don't appear to work without UIDs and I don't believe Native fields generate UIDs in any persistent way. I've been able to get around that in some cases by hard coding a UID for each Native field and Tab I want to use in a condition:

```
$tab = new FieldLayoutTab();
...
$tab->uid = 'SPROUT-UID-CUSTOM-TAB';
$tab->setElements([
  new NativeDropdownField([
    'uid' => 'SPROUT-UID-NATIVE-DROPDOWN-FIELD',
  ]);
]);
```

When I add a Tab condition. It works:

```
$elementCondition = [
  'class' => CustomElementCondition::class,
  'conditionRules' => [
    new DisplayTabConditionRule([
      'value' => true,
    ]),
  ],
];

$tab->setElementCondition($elementCondition);
```

However, when I try to add a condition that acts on two Native Fields, Craft throws an error because it appears to be trying to query the Field with the UID I provide, which doesn't exist, because the field is a Native Field.

```
$elementCondition = [
  'class' => CustomElementCondition::class,
  'conditionRules' => [
    new OptionsFieldConditionRule([
      'uid' => 'SPROUT-UID-NATIVE-TARGET-FIELD',
      'operator' => 'in',
      'values' => ['custom'],
      'fieldUid' => 'SPROUT-UID-NATIVE-DROPDOWN-FIELD',
    ]),
  ],
];

$tab->setElementCondition($elementCondition);
```

As an alternative, I can use a single native field to output both fields that I need and handle the logic for this there, but wondering if it's possible to stay within the native field/condition paradigm to do this now.

**Comment by brandonkelly** [comment, marked as answer]
`OptionsFieldConditionRule` – and anything that implements `FieldConditionRuleInterface` – is specifically for custom fields.  You will need to define your own condition rule class that extends `BaseMultiSelectConditionRule` instead.
